src/main/java/org/commcare/formplayer/services/MediaHandler.kt

```diff
@@ -45,22 +49,24 @@ class MediaHandler(val file: MultipartFile) {
         }
     }
 
-    fun getMediaFilePath(parentDir: String, fileId: String): String {
-        return parentDir + fileId
+    private fun isUnsupportedMimeType(): Boolean {
+        val mimeType: String? = URLConnection.guessContentTypeFromName(file.name)
```

Comment:
since this is just using the file name I think it's basically a duplicate of the extension check

Reply:
That's true in a way although I was assuming that purpose of the mime type check is to cover for the extensions that are not added in our supported extensions list. For eg. even though `webp` is not in our list explicitly, user will still be able to upload the file because it's the valid mimetype. 

Although it seems like we want a more strictier check on content type by checking the byte content of the file and not trust the filename provided by user for the right extension ? I think that should be fine though I think we should still fallback to `guessContentTypeFromName` if we get a null content type from `guessContentTypeFromStream` ?